source/elements/oneMKL/source/domains/dft/config_params/storage_formats.rst

```diff
@@ -7,273 +7,227 @@
-
-
+Depending on the value of the :ref:`onemkl_dft_enum_domain` template value, the
+implementation of the DFT may support different storage schemes for data
+sequences in forward and/or backward domain.
+
+In any case, the data elements are expected to be within contiguous memory
+blocks and defined with generalized strides therein
+(see :ref:`onemkl_dft_config_data_layouts`). For multiple transforms, all data
+sets should be located within the same memory block, and they should be placed
+at the same distance from each other (see
+:ref:`NUMBER_OF TRANSFORMS<onemkl_dft_config_number_of_transforms>`,
+:ref:`onemkl_dft_fwd_bwd_distance`).
 
 .. _onemkl_dft_complex_storage:
 
 COMPLEX_STORAGE
 +++++++++++++++
 
-For the :ref:`onemkl_dft_enum_domain` template parameter with value ``COMPLEX``, both input and output sequences belong to the complex domain. In this case, the configuration parameter COMPLEX_STORAGE can have one of the two values:``COMPLEX_COMPLEX`` (default) or ``REAL_REAL`` .
+For the :ref:`onemkl_dft_enum_domain` template parameter with value ``COMPLEX``,
+data sequences are non-redundant complex sequences in both forward and backward
+domains. In this case, the configuration parameter ``COMPLEX_STORAGE``
+determines how the data is organized in memory, in both forward and backward
+domain. This configuration parameter may take one of the two following values:
+``COMPLEX_COMPLEX`` (default) or ``REAL_REAL``.
 
 .. _onemkl_dft_complex_storage_complex_complex:
 
-.. rubric:: COMPLEX_COMPLEX
-
-With the ``config_value::COMPLEX_COMPLEX`` storage, complex-valued data sequences are stored in a single complex container (array/``sycl::buffer``), AZ, so that a complex-valued element :math:`z_{k_1, k_2,\dots ,k_d}` of the m-th d-dimensional sequence is accessed at :math:`AZ[m*\text{distance} + \text{stride}_0 + k_1*\text{stride}_1 + k_2*\text{stride}_2 + \dots + k_d *\text{stride}_d ]` as a structure consisting of the real and imaginary parts. This code illustrates the use of ``config_value::COMPLEX_COMPLEX`` storage with three dimensions (:math:`n_1,n_2,n_3`) and m batches:
+.. rubric:: COMPLEX_COMPLEX value for configuration parameter COMPLEX_STORAGE
+
+For descriptors of ``COMPLEX`` template :ref:`onemkl_dft_enum_domain` with
+``config_param::COMPLEX_STORAGE`` set to ``config_value::COMPLEX_COMPLEX``,
+forward- and backward-domain data sequences are read/stored from/in a single
```

Comment:
Thank you: revised in [that 221ca79b commit](221ca79bdff7aee1c0571da9aae119a7c9cf9ce9). Give a thumbs-up if that's ok with you.

Reply:
I don't think you've made it clear that forward domain and backward domain data will either share a memory range (for inplace) or not overlap (for not-inplace).

Along with that I'm not too keen on the term "data container"

> data container (device-accessible USM allocation or ``sycl::buffer``)

To me, this implies that each argument to

```
namespace oneapi::mkl::dft {

   template <typename descriptor_type, typename input_type, typename output_type>
   sycl::event compute_forward( descriptor_type                               &desc,
                                input_type                                    *in_re,
                                input_type                                    *in_im,
                                output_type                                   *out_re,
                                output_type                                   *out_im,
                                const std::vector<cl::sycl::event>            &dependencies = {});

}
```
should come from 4 separate calls to `malloc_device` what if you would rather do it all with one big allocation? IMO it would be better to talk about "device accessible memory ranges" or something like that.

Reply:
Regarding
> I don't think you've made it clear that forward domain and backward domain data will either share a memory range (for inplace) or not overlap (for not-inplace).

- I read "The same unique data container is to be used for forward- and backward-domain data sequences for in-place transforms (for descriptor objects with configuration value ``config_value::INPLACE`` for configuration parameter ``config_param::PLACEMENT``)" and a similar statement for ``REAL_REAL`` which, in addition to the example snippets, very much clarifies that (a) shared memory range(s) is(are) to be used for in-place operations, imo. I restructured a little and added clarifications for out-of-place operations in [245b719f327f371519d380e80417108d1a921dc7](https://github.com/oneapi-src/oneAPI-spec/pull/506/commits/245b719f327f371519d380e80417108d1a921dc7), please let me know if you disagree with any of that.

Regarding the use of term "data container" and the ask for rewording to support separate ranges within bigger allocations,
- the term "container" was inherited (and slightly modified) from the current form of the specs;
- as a developer of closed-source oneMKL (and in light of the fact that it is to be made spec-compliant sometime in the future), I would be opposed to changing the wording so because
   1. such usage could break a general alignment assumption which would invalidate most of the GPU implementations;
   2. for sycl buffers, that translates into explicitly stating support for sub-buffers I/O in compute functions which, despite being supported as of now by closed-source oneMKL, may no longer be in the future (to enable targeted optimizations);
- my personal opinion is that this falls beyond the scope of this PR and would deserve a dedicated discussion of its own which should happen afterwards.